pkg/resourceprovider/provider.go

```diff
@@ -398,15 +413,41 @@ func (p *resourceProvider) queryBoth(now pmodel.Time, resource schema.GroupResou
 	wg.Wait()
 
 	if cpuErr != nil {
-		return nsQueryResults{
-			namespace: namespace,
-			err:       fmt.Errorf("unable to fetch node CPU metrics: %v", cpuErr),
+		if promErr, ok := cpuErr.(*client.Error); ok {
+			return nsQueryResults{
+				namespace: namespace,
+				err:      &client.Error{
+					Type: promErr.Type,
+					ErrorMsg:  fmt.Sprintf("unable to fetch node CPU metrics: %s", promErr.ErrorMsg),
```

Comment:
ditto: What's the difference between asserting the type as client.Error vs prom.Error? Is this wrapping promErr with more info?

Reply:
responded in comment above

Reply:
Hm, I think we can just return `promErr` here as the value to the `err` field, instead of reconstructing another `client.Error`. The caller will be the getnodemetrics func anyway, so we'll know it's node cpu metrics related